lib/Access.php

```diff
@@ -1162,36 +1167,51 @@ private function count($filter, $base, $attr = null, $limit = null, $offset = nu
 			 * countEntriesInSearchResults() method signature changed
 			 * by removing $limit and &$hasHitLimit parameters
 			 */
-			$count = $this->countEntriesInSearchResults($sr);
-			$counter += $count;
+			$counts = $this->countEntriesInSearchResults($sr);
+
+			list($hasMorePages, $estimates) = $this->processPagedSearchStatus(
+				$sr, $filter, $base, $count, $limitPerPage,
+				$offset, $pagedSearchOK, $skipHandling
+			);
+
+			foreach ($counts as $i => $count) {
+				$estimate = $estimates[$i];
+				if ($count === $estimate) {
```

Comment:
Is this working as we want? I mean, assuming we have 4 pages, count will have the first 500 entries (if we have a page size of 500) but the estimate is expected to be 1777, for example. The only case where this could be useful is when all the results fits in one page, but in that case we should have already counted all the results in `countEntriesInSearchResults` above.
If the estimate is different than the count, we're still counting all the results.